<a href="https://colab.research.google.com/github/MakC-Ukr/oracle-validation-starknet/blob/master/evaluations/Evaluating_stork.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
import json

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# path_file = "/content/drive/MyDrive/stork_data/BTCUSD5000.json"
# data = json.load(open(path_file))

In [ ]:
# df = pd.DataFrame(data, columns = ['timestamp', 'price', 'other'])
# df.to_csv("/content/drive/MyDrive/stork_data/BTCUSD5000.csv")

In [ ]:
df.tail(2)

timestamp                    price               other
4998  1660282030  23981355000000002654208  339763033386516561
4999  1660282209  24005457455855131688960  339763256843305041

### We have the data for price of **BTC** from Sunday, *July 31*, 2022 4:15:10 AM to Friday, *August 12*, 5:30:09 AM

In [ ]:
df = pd.read_csv("/content/drive/MyDrive/stork_data/BTCUSD5000.csv")
df = df.drop("Unnamed: 0", axis = 1)
df = df.drop("index", axis = 1)
df = df.reset_index()
df = df.drop("index", axis = 1)
df.head(2)

timestamp                    price               other
0  1659240910  23747351837959488798720  338671320287936582
1  1659241090  23767391847961988169728  338671585854488646

In [ ]:
len(df)

5000

In [ ]:
diff_ls = []
diff = 0
prev = 0
for ind, row in df.iterrows():
  if prev == 0:
    prev = row['timestamp']
    continue
  curr_diff = (row['timestamp'] - prev)
  diff += curr_diff
  diff_ls.append(curr_diff)
  prev = row['timestamp']

print("the avergae differenc in seconds between two subsequent rows is :", pd.Series(diff_ls).median())

the avergae differenc in seconds between two subsequent rows is : 180.0


In [ ]:
sr = pd.Series(diff_ls)

180.0

In [ ]:
sr[sr > 500]

416       541
417      1980
447      3292
2324      540
2325      540
2326      544
2327      537
2328      539
2343    15121
2344      544
3207      541
3217      542
3218    19981
3431      542
3432      537
3434      540
3435      541
3436      541
3438      543
3439      538
3440      539
4598      541
4600      541
4602      540
4615      540
4616    20777
dtype: int64

In [ ]:
df[2343: 2345]

timestamp                    price               other
2343  1659692531  23144621693253990744064  339145173766242374
2344  1659707652  23026996349817489195008  339161094985613383

**Multiple instances in a 12 day timeframe when there was no report of price for more than 5 minutes. 
Sometimes the outage was for more than 15,000 seconds (> 4 hours)**

In [ ]:
new_df = []

for ind, row in df.iterrows():
  ls = {}
  ls['timestamp'] = row['timestamp']
  ls['price'] = row['price'][:8]
  new_df.append(ls)

new_df = pd.DataFrame(new_df)
new_df.head(2)

timestamp     price
0  1659240910  23747351
1  1659241090  23767391

In [ ]:
new_df['price']= pd.to_numeric(new_df['price'])

In [ ]:
new_df['price']= new_df['price'].div(1000)

In [ ]:
new_df.head(2)

timestamp      price
0  1659240910  23747.351
1  1659241090  23767.391

In [ ]:
print(new_df.describe()['price']['max'] - new_df.describe()['price']['min'])

2403.268


**Range of price for BTC was:  2403.268. Which sounds fine**


### We have the data for price of **ETH** from Sunday, *July 31*, 2022 4:15:10 AM to Friday, *August 12*, 5:30:09 AM

In [ ]:
# path_file = "/content/drive/MyDrive/stork_data/ETHUSD5000.json"
# data = json.load(open(path_file))
# df = pd.DataFrame(data, columns = ['timestamp', 'price', 'other'])
# df.sort_values(by="timestamp", inplace = True)
# new_df = []
# for ind, row in df.iterrows():
#   ls = {}
#   ls['timestamp'] = row['timestamp']
#   ls['price'] = row['price'][:8]
#   new_df.append(ls)

# new_df = pd.DataFrame(new_df)
# new_df['price']= pd.to_numeric(new_df['price'])
# new_df['price']= new_df['price'].div(10000)
# new_df.to_csv("/content/drive/MyDrive/stork_data/ETHUSD5000.csv")

In [ ]:
new_df = pd.read_csv("/content/drive/MyDrive/stork_data/ETHUSD5000.csv")

In [ ]:
diff_ls = []
diff = 0
prev = 0
for ind, row in new_df.iterrows():
  if prev == 0:
    prev = row['timestamp']
    continue
  curr_diff = (row['timestamp'] - prev)
  diff += curr_diff
  diff_ls.append(curr_diff)
  prev = row['timestamp']

print("the avergae difference in seconds between two subsequent rows is :", pd.Series(diff_ls).median())

the avergae differenc in seconds between two subsequent rows is : 180.0


In [ ]:
sr = pd.Series(diff_ls)

In [ ]:
sr[sr>500]

416       541.0
417      1980.0
447      3292.0
2324      540.0
2325      540.0
2326      544.0
2327      537.0
2328      539.0
2343    15121.0
2344      544.0
3207      541.0
3217      542.0
3218    19981.0
3431      542.0
3432      537.0
3434      540.0
3435      541.0
3436      541.0
3438      543.0
3439      538.0
3440      539.0
4598      541.0
4600      541.0
4602      540.0
4615      540.0
4616    20777.0
dtype: float64